### ISSUE TITLE
Crayon Shin-chan - Arashi o Yobu Enji (Japan) - missing fog effect [Fast PPU on]

### ISSUE THREAD

**Issue by Max833** [byuu/bsnes#73]
Second level of  attract mode.

![Crayon Shin-chan - Arashi o Yobu Enji (Japan)](https://user-images.githubusercontent.com/53529402/64041696-744ccd00-cb60-11e9-9fa2-a03b7e0d59fb.jpg)

bsnes v075:
![image](https://user-images.githubusercontent.com/53529402/64043939-effd4880-cb65-11e9-879f-259c75a4e48c.png)

Hardware (doesn't look right either...?):
![aa](https://user-images.githubusercontent.com/53529402/64041713-80388f00-cb60-11e9-8789-2efe2daa8795.jpeg)

**Comment by Max833** [comment]
Behaviour is not really consistent.

"Fast PPU on" can look like this:

![image](https://user-images.githubusercontent.com/53529402/64055456-d02d4b00-cb8c-11e9-83f8-8d6523d69bee.png)

And "Fast PPU off":

![image](https://user-images.githubusercontent.com/53529402/64055567-842ed600-cb8d-11e9-98bd-2ec33b7634c1.png)

**Comment by byuu** [comment]
Since it's such an obscure title: rather than risk a major speed hit or breaking other games, I chose to disable the fast PPU for this title to get the color blending to work correctly.

As for the accurate PPU, this is what the game is doing:

2109 @ 48, 958 BG3SC (0x2800 -> 0x3800)
210c @ 48,1012 BG34NBA
212c @ 48,1066 TM
212d @ 48,1120 TS
2131 @ 48,1174 CGADDSUB
2106 @ 48,1228 MOSAIC
2111 @ 48,1304 BG3HOFS
2111 @ 48,1356 BG3HOFS
2112 @ 49,  44 BG3VOFS
2112 @ 49,  96 BG3VOFS

It's the write to $2109 while the screen is rendering that causes the stray bad pixels. I added a BGnSC latch for now, but the longer-term fix is to latch the fetched tile from VRAM, which I'll work on now. Anyway, the latest commit onward should solve it.

Your screenshot doesn't show me if Mesen-S shows those bad pixels at the top right or not due to the position of the clouds. The other issue is just Mesen-S not blending pseudo-hires layers (both hires and pseudo-hires should be blended.)
